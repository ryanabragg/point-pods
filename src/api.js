import localforage from 'localforage';
import React from 'react';

const api = {
  store: localforage.createInstance({
    name: 'Point Pods',
    storeName: 'point_pods',
    description: 'some description'
  }),
};

export default api;
export const APIContext = React.createContext(api);
export function withAPI(Component) {
  return function APIComponent(props) {
    return (
      <APIContext.Consumer>
        {context => <Component {...props} api={context} />}
      </APIContext.Consumer>
    );
  };
}
