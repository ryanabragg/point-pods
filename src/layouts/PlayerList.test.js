import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Dialog from '@material-ui/core/Dialog';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';

import PlayerList from './PlayerList';

import api from '../api';

import { testPlayers, mockStorage, mockAPI, demockAPI } from '../testData';

beforeEach(() => mockAPI(api, mockStorage));
afterEach(() => demockAPI(api));

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<PlayerList />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('matches the prior snapshot', (done) => {
  const component = renderer.create(<PlayerList />);
  expect(component.toJSON()).toMatchSnapshot();
  component.update(<PlayerList />);
  setTimeout(() => {
    try {
      expect(component.toJSON()).toMatchSnapshot();
      done();
    } catch (error) {
      done.fail(error);
    }
  }, 100);
});

test('is a HOC-ed component', () => {
  const wrapper = shallow(<PlayerList />);
  expect(wrapper.type().prototype.constructor.name).toBe('APIComponent');
  expect(wrapper.dive().type().prototype.constructor.name).toBe('PlayerList');
});

describe('defaults', () => {
  test('values', () => {
    const wrapper = shallow(<PlayerList />);
    const component = wrapper.dive().dive();
    expect(component.state('dialog')).toBe(null);
    expect(component.state('searching')).toBe(false);
    expect(component.state('sort')).toEqual({field: 'name', ascending: true});
    expect(component.state('selected')).toEqual([]);
  });

  test('loads the state', (done) => {
    const wrapper = shallow(<PlayerList />);
    const component = wrapper.dive().dive();
    expect(component.state('players')).toEqual([]);
    setTimeout(() => {
      try {
        expect(component.state('players')).toEqual(testPlayers);
        done();
      } catch (error) {
        done.fail(error);
      }
    }, 100);
  });
});

describe('actions', () => {
  test('select');

  test('searching');

  test('sorting');

  test('add a player', (done) => {
    const spy = jest.fn((m, v, d, c) => null);
    const wrapper = shallow(<PlayerList notification={spy} />);
    const component = wrapper.dive().dive();
    expect(component.find(Dialog).length).toBe(0);
    expect(component.state('dialog')).toBe(null);
    setTimeout(() => {
      try {
        component.findWhere(node => node.prop('variant') === 'fab').simulate('click');
        expect(component.state('dialog')).toBe('new-player');
        expect(component.find(Dialog).length).toBe(1);
        component.find('#new-player-cancel').simulate('click');
        expect(component.find(Dialog).length).toBe(0);
        expect(component.state('dialog')).toBe(null);
        component.findWhere(node => node.prop('variant') === 'fab').simulate('click');
        const newName = 'Add Test';
        component.find('#new-player-name').simulate('change', {target: {value: newName}});
        component.find('#new-player-submit').simulate('click');
        setTimeout(() => {
          try {
            const last = component.state('players').length - 1;
            expect(component.state('players')[last].name).toBe(newName);
            expect(spy.mock.calls[0][0]).toBe(`Created player ${newName}`);
            expect(spy.mock.calls[0][1]).toBe(`info`);
            done();
          } catch(error) {
            done.fail(error);
          }
        }, 100);
      } catch(error) {
        done.fail(error);
      }
    }, 100);
  });
});
