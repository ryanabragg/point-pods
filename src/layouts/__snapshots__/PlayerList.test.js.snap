// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the prior snapshot 1`] = `
<div>
  <div
    className="APIComponent-root-165"
  >
    <header
      className="MuiPaper-root-181 MuiPaper-elevation4-187 MuiAppBar-root-173 MuiAppBar-positionFixed-174 MuiAppBar-colorPrimary-179 mui-fixed"
    >
      <div
        className="MuiToolbar-root-208 MuiToolbar-regular-210 MuiToolbar-gutters-209 APIComponent-nowrap-167"
      >
        <button
          aria-label="Menu"
          className="MuiButtonBase-root-218 MuiIconButton-root-212 MuiIconButton-colorInherit-213 APIComponent-menuButton-168"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-217"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-221"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-228"
          />
        </button>
        <h2
          className="MuiTypography-root-235 MuiTypography-title-241 MuiTypography-colorInherit-254 APIComponent-flex-166"
        >
          Point Pods
        </h2>
        <button
          className="MuiButtonBase-root-218 MuiIconButton-root-212 MuiIconButton-colorInherit-213"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-217"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-221"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-228"
          />
        </button>
        <div>
          <button
            aria-haspopup="true"
            aria-owns={null}
            className="MuiButtonBase-root-218 MuiIconButton-root-212 MuiIconButton-colorInherit-213"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiIconButton-label-217"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-221"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-228"
            />
          </button>
        </div>
      </div>
    </header>
    <div
      className="APIComponent-spacer-170"
    />
  </div>
  <button
    aria-label="New Player"
    className="MuiButtonBase-root-218 MuiButton-root-285 MuiButton-fab-300 MuiButton-contained-294 MuiButton-containedSecondary-296 MuiButton-raised-297 MuiButton-raisedSecondary-299 APIComponent-actionButton-160"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-286"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-221"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-228"
    />
  </button>
  <div
    className="PlayerCards-root-309"
  >
    <div
      className="MuiPaper-root-181 MuiPaper-elevation1-184 MuiPaper-rounded-182 MuiCard-root-312 PlayerCards-card-310"
      id="not-found"
    >
      <div
        className="MuiCardHeader-root-313"
      >
        <div
          className="MuiCardHeader-content-316"
        >
          <span
            className="MuiTypography-root-235 MuiTypography-headline-240 MuiCardHeader-title-317"
          >
            No Players Found
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`matches the prior snapshot 2`] = `
<div>
  <div
    className="APIComponent-root-165"
  >
    <header
      className="MuiPaper-root-181 MuiPaper-elevation4-187 MuiAppBar-root-173 MuiAppBar-positionFixed-174 MuiAppBar-colorPrimary-179 mui-fixed"
    >
      <div
        className="MuiToolbar-root-208 MuiToolbar-regular-210 MuiToolbar-gutters-209 APIComponent-nowrap-167"
      >
        <button
          aria-label="Menu"
          className="MuiButtonBase-root-218 MuiIconButton-root-212 MuiIconButton-colorInherit-213 APIComponent-menuButton-168"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-217"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-221"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-228"
          />
        </button>
        <h2
          className="MuiTypography-root-235 MuiTypography-title-241 MuiTypography-colorInherit-254 APIComponent-flex-166"
        >
          Point Pods
        </h2>
        <button
          className="MuiButtonBase-root-218 MuiIconButton-root-212 MuiIconButton-colorInherit-213"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-217"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-221"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-228"
          />
        </button>
        <div>
          <button
            aria-haspopup="true"
            aria-owns={null}
            className="MuiButtonBase-root-218 MuiIconButton-root-212 MuiIconButton-colorInherit-213"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiIconButton-label-217"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-221"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-228"
            />
          </button>
        </div>
      </div>
    </header>
    <div
      className="APIComponent-spacer-170"
    />
  </div>
  <button
    aria-label="New Player"
    className="MuiButtonBase-root-218 MuiButton-root-285 MuiButton-fab-300 MuiButton-contained-294 MuiButton-containedSecondary-296 MuiButton-raised-297 MuiButton-raisedSecondary-299 APIComponent-actionButton-160"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-286"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-221"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-228"
    />
  </button>
  <div
    className="PlayerCards-root-309"
  >
    <div
      className="MuiPaper-root-181 MuiPaper-elevation1-184 MuiPaper-rounded-182 MuiCard-root-312 PlayerCards-card-310"
      id="mock0"
    >
      <div
        className="MuiCardHeader-root-313"
      >
        <div
          className="MuiCardHeader-content-316"
        >
          <span
            className="MuiTypography-root-235 MuiTypography-headline-240 MuiCardHeader-title-317"
          >
            test0
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root-319"
      >
        <p
          className="MuiTypography-root-235 MuiTypography-body1-244"
        >
          0 Cumulative Points
        </p>
      </div>
      <div
        className="MuiCardActions-root-320"
      >
        <a
          className="MuiButtonBase-root-218 MuiButton-root-285 MuiButton-text-287 MuiButton-flat-290 MuiButton-sizeSmall-306 PlayerCards-right-311 MuiCardActions-action-321"
          href="/player/mock0"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-286"
          >
            Details
          </span>
          <span
            className="MuiTouchRipple-root-228"
          />
        </a>
      </div>
    </div>
    <div
      className="MuiPaper-root-181 MuiPaper-elevation1-184 MuiPaper-rounded-182 MuiCard-root-312 PlayerCards-card-310"
      id="mock1"
    >
      <div
        className="MuiCardHeader-root-313"
      >
        <div
          className="MuiCardHeader-content-316"
        >
          <span
            className="MuiTypography-root-235 MuiTypography-headline-240 MuiCardHeader-title-317"
          >
            test1
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root-319"
      >
        <p
          className="MuiTypography-root-235 MuiTypography-body1-244"
        >
          1 Cumulative Points
        </p>
      </div>
      <div
        className="MuiCardActions-root-320"
      >
        <a
          className="MuiButtonBase-root-218 MuiButton-root-285 MuiButton-text-287 MuiButton-flat-290 MuiButton-sizeSmall-306 PlayerCards-right-311 MuiCardActions-action-321"
          href="/player/mock1"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-286"
          >
            Details
          </span>
          <span
            className="MuiTouchRipple-root-228"
          />
        </a>
      </div>
    </div>
    <div
      className="MuiPaper-root-181 MuiPaper-elevation1-184 MuiPaper-rounded-182 MuiCard-root-312 PlayerCards-card-310"
      id="mock2"
    >
      <div
        className="MuiCardHeader-root-313"
      >
        <div
          className="MuiCardHeader-content-316"
        >
          <span
            className="MuiTypography-root-235 MuiTypography-headline-240 MuiCardHeader-title-317"
          >
            test2
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root-319"
      >
        <p
          className="MuiTypography-root-235 MuiTypography-body1-244"
        >
          2 Cumulative Points
        </p>
      </div>
      <div
        className="MuiCardActions-root-320"
      >
        <a
          className="MuiButtonBase-root-218 MuiButton-root-285 MuiButton-text-287 MuiButton-flat-290 MuiButton-sizeSmall-306 PlayerCards-right-311 MuiCardActions-action-321"
          href="/player/mock2"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-286"
          >
            Details
          </span>
          <span
            className="MuiTouchRipple-root-228"
          />
        </a>
      </div>
    </div>
    <div
      className="MuiPaper-root-181 MuiPaper-elevation1-184 MuiPaper-rounded-182 MuiCard-root-312 PlayerCards-card-310"
      id="mock3"
    >
      <div
        className="MuiCardHeader-root-313"
      >
        <div
          className="MuiCardHeader-content-316"
        >
          <span
            className="MuiTypography-root-235 MuiTypography-headline-240 MuiCardHeader-title-317"
          >
            test3
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root-319"
      >
        <p
          className="MuiTypography-root-235 MuiTypography-body1-244"
        >
          3 Cumulative Points
        </p>
      </div>
      <div
        className="MuiCardActions-root-320"
      >
        <a
          className="MuiButtonBase-root-218 MuiButton-root-285 MuiButton-text-287 MuiButton-flat-290 MuiButton-sizeSmall-306 PlayerCards-right-311 MuiCardActions-action-321"
          href="/player/mock3"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-286"
          >
            Details
          </span>
          <span
            className="MuiTouchRipple-root-228"
          />
        </a>
      </div>
    </div>
    <div
      className="MuiPaper-root-181 MuiPaper-elevation1-184 MuiPaper-rounded-182 MuiCard-root-312 PlayerCards-card-310"
      id="mock4"
    >
      <div
        className="MuiCardHeader-root-313"
      >
        <div
          className="MuiCardHeader-content-316"
        >
          <span
            className="MuiTypography-root-235 MuiTypography-headline-240 MuiCardHeader-title-317"
          >
            test4
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root-319"
      >
        <p
          className="MuiTypography-root-235 MuiTypography-body1-244"
        >
          4 Cumulative Points
        </p>
      </div>
      <div
        className="MuiCardActions-root-320"
      >
        <a
          className="MuiButtonBase-root-218 MuiButton-root-285 MuiButton-text-287 MuiButton-flat-290 MuiButton-sizeSmall-306 PlayerCards-right-311 MuiCardActions-action-321"
          href="/player/mock4"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-286"
          >
            Details
          </span>
          <span
            className="MuiTouchRipple-root-228"
          />
        </a>
      </div>
    </div>
    <div
      className="MuiPaper-root-181 MuiPaper-elevation1-184 MuiPaper-rounded-182 MuiCard-root-312 PlayerCards-card-310"
      id="mock5"
    >
      <div
        className="MuiCardHeader-root-313"
      >
        <div
          className="MuiCardHeader-content-316"
        >
          <span
            className="MuiTypography-root-235 MuiTypography-headline-240 MuiCardHeader-title-317"
          >
            test5
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root-319"
      >
        <p
          className="MuiTypography-root-235 MuiTypography-body1-244"
        >
          5 Cumulative Points
        </p>
      </div>
      <div
        className="MuiCardActions-root-320"
      >
        <a
          className="MuiButtonBase-root-218 MuiButton-root-285 MuiButton-text-287 MuiButton-flat-290 MuiButton-sizeSmall-306 PlayerCards-right-311 MuiCardActions-action-321"
          href="/player/mock5"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-286"
          >
            Details
          </span>
          <span
            className="MuiTouchRipple-root-228"
          />
        </a>
      </div>
    </div>
    <div
      className="MuiPaper-root-181 MuiPaper-elevation1-184 MuiPaper-rounded-182 MuiCard-root-312 PlayerCards-card-310"
      id="mock6"
    >
      <div
        className="MuiCardHeader-root-313"
      >
        <div
          className="MuiCardHeader-content-316"
        >
          <span
            className="MuiTypography-root-235 MuiTypography-headline-240 MuiCardHeader-title-317"
          >
            test6
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root-319"
      >
        <p
          className="MuiTypography-root-235 MuiTypography-body1-244"
        >
          6 Cumulative Points
        </p>
      </div>
      <div
        className="MuiCardActions-root-320"
      >
        <a
          className="MuiButtonBase-root-218 MuiButton-root-285 MuiButton-text-287 MuiButton-flat-290 MuiButton-sizeSmall-306 PlayerCards-right-311 MuiCardActions-action-321"
          href="/player/mock6"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-286"
          >
            Details
          </span>
          <span
            className="MuiTouchRipple-root-228"
          />
        </a>
      </div>
    </div>
    <div
      className="MuiPaper-root-181 MuiPaper-elevation1-184 MuiPaper-rounded-182 MuiCard-root-312 PlayerCards-card-310"
      id="mock7"
    >
      <div
        className="MuiCardHeader-root-313"
      >
        <div
          className="MuiCardHeader-content-316"
        >
          <span
            className="MuiTypography-root-235 MuiTypography-headline-240 MuiCardHeader-title-317"
          >
            test7
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root-319"
      >
        <p
          className="MuiTypography-root-235 MuiTypography-body1-244"
        >
          7 Cumulative Points
        </p>
      </div>
      <div
        className="MuiCardActions-root-320"
      >
        <a
          className="MuiButtonBase-root-218 MuiButton-root-285 MuiButton-text-287 MuiButton-flat-290 MuiButton-sizeSmall-306 PlayerCards-right-311 MuiCardActions-action-321"
          href="/player/mock7"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-286"
          >
            Details
          </span>
          <span
            className="MuiTouchRipple-root-228"
          />
        </a>
      </div>
    </div>
    <div
      className="MuiPaper-root-181 MuiPaper-elevation1-184 MuiPaper-rounded-182 MuiCard-root-312 PlayerCards-card-310"
      id="mock8"
    >
      <div
        className="MuiCardHeader-root-313"
      >
        <div
          className="MuiCardHeader-content-316"
        >
          <span
            className="MuiTypography-root-235 MuiTypography-headline-240 MuiCardHeader-title-317"
          >
            test8
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root-319"
      >
        <p
          className="MuiTypography-root-235 MuiTypography-body1-244"
        >
          8 Cumulative Points
        </p>
      </div>
      <div
        className="MuiCardActions-root-320"
      >
        <a
          className="MuiButtonBase-root-218 MuiButton-root-285 MuiButton-text-287 MuiButton-flat-290 MuiButton-sizeSmall-306 PlayerCards-right-311 MuiCardActions-action-321"
          href="/player/mock8"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-286"
          >
            Details
          </span>
          <span
            className="MuiTouchRipple-root-228"
          />
        </a>
      </div>
    </div>
    <div
      className="MuiPaper-root-181 MuiPaper-elevation1-184 MuiPaper-rounded-182 MuiCard-root-312 PlayerCards-card-310"
      id="mock9"
    >
      <div
        className="MuiCardHeader-root-313"
      >
        <div
          className="MuiCardHeader-content-316"
        >
          <span
            className="MuiTypography-root-235 MuiTypography-headline-240 MuiCardHeader-title-317"
          >
            test9
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root-319"
      >
        <p
          className="MuiTypography-root-235 MuiTypography-body1-244"
        >
          9 Cumulative Points
        </p>
      </div>
      <div
        className="MuiCardActions-root-320"
      >
        <a
          className="MuiButtonBase-root-218 MuiButton-root-285 MuiButton-text-287 MuiButton-flat-290 MuiButton-sizeSmall-306 PlayerCards-right-311 MuiCardActions-action-321"
          href="/player/mock9"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-286"
          >
            Details
          </span>
          <span
            className="MuiTouchRipple-root-228"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
